### ISSUE TITLE
r/aws_lightsail_instance: Crash when adding tag with empty value

### ISSUE THREAD

**Issue by ewbankkit** [terraform-providers/terraform-provider-aws#12368]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

AWS provider v2.52.0.

### Affected Resource(s)

* `aws_lightsail_instance`

### Terraform Configuration Files

```hcl
resource "aws_lightsail_instance" "lightsail_instance_test" {
  name              = "terraform-test"
  availability_zone = "${data.aws_availability_zones.available.names[0]}"
  blueprint_id      = "amazon_linux"
  bundle_id         = "nano_1_0"

  tags = {
    Name = "terraform-test"
    KeyOnlyTag = ""
  }
}
```

### Panic Output

```
aws_lightsail_instance.lightsail_instance_test: Still creating... (10s elapsed)
aws_lightsail_instance.lightsail_instance_test: Still creating... (20s elapsed)

Error: Error applying plan:

1 error occurred:
	* aws_lightsail_instance.lightsail_instance_test: 1 error occurred:
	* aws_lightsail_instance.lightsail_instance_test: unexpected EOF





Terraform does not automatically rollback in the face of errors.
Instead, your Terraform state file has been partially updated with
any resources that successfully completed. Please address the error
above and apply again to incrementally change your infrastructure.


panic: runtime error: invalid memory address or nil pointer dereference
2020-03-12T09:27:00.257-0400 [DEBUG] plugin.terraform-provider-aws: [signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x384f82a]
2020-03-12T09:27:00.257-0400 [DEBUG] plugin.terraform-provider-aws: 
2020-03-12T09:27:00.257-0400 [DEBUG] plugin.terraform-provider-aws: goroutine 257 [running]:
2020-03-12T09:27:00.257-0400 [DEBUG] plugin.terraform-provider-aws: github.com/terraform-providers/terraform-provider-aws/aws/internal/keyvaluetags.KeyValueTags.Map(0xc000a679b0, 0xc000a679b0)
2020-03-12T09:27:00.257-0400 [DEBUG] plugin.terraform-provider-aws: 	/home/kit/wrk/src/github.com/terraform-providers/terraform-provider-aws/aws/internal/keyvaluetags/key_value_tags.go:138 +0xda
2020-03-12T09:27:00.257-0400 [DEBUG] plugin.terraform-provider-aws: github.com/terraform-providers/terraform-provider-aws/aws.resourceAwsLightsailInstanceRead(0xc00039e150, 0x4df1960, 0xc000ee2500, 0x0, 0x0)
2020-03-12T09:27:00.257-0400 [DEBUG] plugin.terraform-provider-aws: 	/home/kit/wrk/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_lightsail_instance.go:216 +0x6eb
2020-03-12T09:27:00.257-0400 [DEBUG] plugin.terraform-provider-aws: github.com/terraform-providers/terraform-provider-aws/aws.resourceAwsLightsailInstanceCreate(0xc00039e150, 0x4df1960, 0xc000ee2500, 0xc00039e150, 0x0)
2020-03-12T09:27:00.258-0400 [DEBUG] plugin.terraform-provider-aws: 	/home/kit/wrk/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_lightsail_instance.go:170 +0x723
2020-03-12T09:27:00.258-0400 [DEBUG] plugin.terraform-provider-aws: github.com/hashicorp/terraform-plugin-sdk/helper/schema.(*Resource).Apply(0xc000a34b00, 0xc0001cd860, 0xc0005c1a60, 0x4df1960, 0xc000ee2500, 0xc0006ce001, 0xc000075ba8, 0x40da86)
2020-03-12T09:27:00.258-0400 [DEBUG] plugin.terraform-provider-aws: 	/home/kit/wrk/pkg/mod/github.com/hashicorp/terraform-plugin-sdk@v1.7.0/helper/schema/resource.go:305 +0x365
2020-03-12T09:27:00.258-0400 [DEBUG] plugin.terraform-provider-aws: github.com/hashicorp/terraform-plugin-sdk/helper/schema.(*Provider).Apply(0xc000c19480, 0xc000bb8040, 0xc0001cd860, 0xc0005c1a60, 0xc000d15a70, 0xc000ce7ef0, 0x18)
2020-03-12T09:27:00.258-0400 [DEBUG] plugin.terraform-provider-aws: 	/home/kit/wrk/pkg/mod/github.com/hashicorp/terraform-plugin-sdk@v1.7.0/helper/schema/provider.go:294 +0x99
2020-03-12T09:27:00.258-0400 [DEBUG] plugin.terraform-provider-aws: github.com/hashicorp/terraform-plugin-sdk/plugin.(*ResourceProviderServer).Apply(0xc00052bd20, 0xc0005c14a0, 0xc000530400, 0x0, 0x0)
2020-03-12T09:27:00.258-0400 [DEBUG] plugin.terraform-provider-aws: 	/home/kit/wrk/pkg/mod/github.com/hashicorp/terraform-plugin-sdk@v1.7.0/plugin/resource_provider.go:530 +0x57
2020-03-12T09:27:00.258-0400 [DEBUG] plugin.terraform-provider-aws: reflect.Value.call(0xc000550cc0, 0xc000bcdfe8, 0x13, 0x58b4395, 0x4, 0xc000075f18, 0x3, 0x3, 0x406e9a, 0xc0003b7560, ...)
2020-03-12T09:27:00.258-0400 [DEBUG] plugin.terraform-provider-aws: 	/usr/local/go/src/reflect/value.go:460 +0x5f6
2020-03-12T09:27:00.258-0400 [DEBUG] plugin.terraform-provider-aws: reflect.Value.Call(0xc000550cc0, 0xc000bcdfe8, 0x13, 0xc000846718, 0x3, 0x3, 0x67569b, 0xc000ff8000, 0x0)
2020-03-12T09:27:00.258-0400 [DEBUG] plugin.terraform-provider-aws: 	/usr/local/go/src/reflect/value.go:321 +0xb4
2020-03-12T09:27:00.258-0400 [DEBUG] plugin.terraform-provider-aws: net/rpc.(*service).call(0xc000c9a8c0, 0xc0005179f0, 0xc0004ca0b0, 0xc0004ca0c0, 0xc000c19500, 0xc0000c1460, 0x460e900, 0xc0005c14a0, 0x16, 0x460e940, ...)
2020-03-12T09:27:00.258-0400 [DEBUG] plugin.terraform-provider-aws: 	/usr/local/go/src/net/rpc/server.go:377 +0x16f
2020-03-12T09:27:00.258-0400 [DEBUG] plugin.terraform-provider-aws: created by net/rpc.(*Server).ServeCodec
2020-03-12T09:27:00.258-0400 [DEBUG] plugin.terraform-provider-aws: 	/usr/local/go/src/net/rpc/server.go:474 +0x42b
```

### Steps to Reproduce

1. `terraform apply`

### References

First noticed by @stefansundin in https://github.com/terraform-providers/terraform-provider-aws/pull/12346.

Relevant code:

https://github.com/terraform-providers/terraform-provider-aws/blob/654cc23a38444d4836155d6a2243e3976b5f3992/aws/internal/keyvaluetags/key_value_tags.go#L134-L142

**Comment by ewbankkit** [comment]
The behaviour of empty tag values differs between AWS services, e.g. for EC2/VPC the equivalent does not crash:

```hcl
resource "aws_vpc" "test" {
	cidr_block = "10.1.0.0/16"
	tags = {
		Name = "terraform-test"
    KeyOnlyTag = ""
	}
}
```

**Comment by bflad** [comment]
The fix for this has been merged and will release with version 3.5.0 of the Terraform AWS Provider, later this week.
